bpo-45223: Lib/pty.py: handle stdin I/O errors same way as master I/O errors

kanavin opened the pull request:
reading stdin can throw the same I/O errors as reading from master fd does,
e.g. when running under Yocto's test harness:
======================================================================
ERROR: test_spawn_doesnt_hang (test.test_pty.PtyTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.10/test/test_pty.py", line 316, in test_spawn_doesnt_hang
    pty.spawn([sys.executable, '-c', 'print("hi there")'])
  File "/usr/lib/python3.10/pty.py", line 181, in spawn
    _copy(master_fd, master_read, stdin_read)
  File "/usr/lib/python3.10/pty.py", line 157, in _copy
    data = stdin_read(STDIN_FILENO)
  File "/usr/lib/python3.10/pty.py", line 132, in _read
    return os.read(fd, 1024)
OSError: [Errno 5] Input/output error

So let's treat both channels the same.


Signed-off-by: Alexander Kanavin <alex@linutronix.de>


<!-- issue-number: [bpo-45223](https://bugs.python.org/issue45223) -->
https://bugs.python.org/issue45223
<!-- /issue-number -->

MaxwellDupre reviewed (approved):
Looks good:
Ran 7 tests in 0.049s

OK (expected failures=1)

== Tests result: SUCCESS ==

kanavin committed:
Lib/pty.py: handle stdin I/O errors same way as master I/O errors

reading stdin can throw the same I/O errors as reading from master fd does,
e.g. when running under Yocto's test harness:
======================================================================
ERROR: test_spawn_doesnt_hang (test.test_pty.PtyTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.10/test/test_pty.py", line 316, in test_spawn_doesnt_hang
    pty.spawn([sys.executable, '-c', 'print("hi there")'])
  File "/usr/lib/python3.10/pty.py", line 181, in spawn
    _copy(master_fd, master_read, stdin_read)
  File "/usr/lib/python3.10/pty.py", line 157, in _copy
    data = stdin_read(STDIN_FILENO)
  File "/usr/lib/python3.10/pty.py", line 132, in _read
    return os.read(fd, 1024)
OSError: [Errno 5] Input/output error

So let's treat both channels the same.

Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

blurb-it[bot] committed:
📜🤖 Added by blurb_it.
